src/final_chain/final_chain.cpp

```diff
@@ -0,0 +1,440 @@
+#include "final_chain.hpp"
+
+#include "TrieCommon.h"
+#include "common/constants.hpp"
+#include "replay_protection_service.hpp"
+#include "util/thread_pool.hpp"
+
+namespace taraxa::final_chain {
+using namespace std;
+using namespace dev;
+
+enum DBMetaKeys { LAST_NUMBER = 1 };
+
+struct FinalChainImpl final : FinalChain {
+  shared_ptr<DB> db_;
+  StateAPI state_api;
+  unique_ptr<ReplayProtectionService> replay_protection_service_{new ReplayProtectionServiceDummy};
```

Comment:
please fix this. Using "new" is just huge NO

Reply:
ok

Reply:
done